### ISSUE TITLE
'Unable to load shared library 'kernel32.dll' or one of its dependencies.

### ISSUE THREAD

**Issue by khteh** [dotnet/aspnetcore#41417]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Start debugging session on my ASP.Net 6 application from VSCode on Ubuntu 22.04 bumps into the following exception:
```
Exception has occurred: CLR/System.DllNotFoundException
Exception thrown: 'System.DllNotFoundException' in Microsoft.AspNetCore.Cryptography.Internal.dll: 'Unable to load shared library 'kernel32.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libkernel32.dll: cannot open shared object file: No such file or directory'
   at Microsoft.AspNetCore.Cryptography.SafeHandles.SafeLibraryHandle.UnsafeNativeMethods.LoadLibraryEx(String lpFileName, IntPtr hFile, UInt32 dwFlags)
   at Microsoft.AspNetCore.Cryptography.SafeHandles.SafeLibraryHandle.Open(String filename)
```

### Expected Behavior

No exception

### Exceptions (if any)

```
Exception has occurred: CLR/System.DllNotFoundException
Exception thrown: 'System.DllNotFoundException' in Microsoft.AspNetCore.Cryptography.Internal.dll: 'Unable to load shared library 'kernel32.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libkernel32.dll: cannot open shared object file: No such file or directory'
   at Microsoft.AspNetCore.Cryptography.SafeHandles.SafeLibraryHandle.UnsafeNativeMethods.LoadLibraryEx(String lpFileName, IntPtr hFile, UInt32 dwFlags)
   at Microsoft.AspNetCore.Cryptography.SafeHandles.SafeLibraryHandle.Open(String filename)
```

### .NET Version

6.0.202

### Anything else?

```
.NET SDK (reflecting any global.json):
 Version:   6.0.202
 Commit:    f8a55617d2

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         ubuntu.22.04-x64
 Base Path:   /usr/share/dotnet/sdk/6.0.202/

Host (useful for support):
  Version: 6.0.4
  Commit:  be98e88c76

.NET SDKs installed:
  6.0.202 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by adityamandaleeka** [comment]
Can you share more info with exact repro steps? Ideally if you could push your project to GitHub so we could try it that would be great.

**Comment by khteh** [comment]
https://github.com/khteh/AspNetCoreApiStarter

**Comment by halter73** [comment]
Thanks @khteh! After switching from the `/tmp/localhost.pfx` file to the default developer certificate, I was able to see the issue. I opened a PR to fix the issue at #41487.

Since this appears to only be a first-chance exception, I'm not sure we're going to patch .NET 6 to fix this.
